### DISCUSSION TITLE
Shared Plugin directory/storage option

### DISCUSSION THREAD

**Discussion by awgymer** [nextflow-io/nextflow#4245] [Q&A]
## New feature

When downloading plugins nextflow stores them under the `$HOME/.nextflow` directory. It would be useful to allow for a single shared location. 

## Usage scenario 

The use-case here is on an HPC where the external connection to GitHub is locked down or restricted or the admins wish to control the content downloaded. This amounts to using the plugins in "offline" mode essentially. Currently we would be required to copy plugin code to each individuals home space.

The reason we don't want to simply change the `NXF_HOME` variable is that pulling pipelines from internally hosted/managed version control is still possible and we don't to affect the other user-specific content in that location. 

There is some further discussion on the Slack on this issue [here](https://nextflow.slack.com/archives/C02T98BKE2K/p1693393236048169)

**Comment by pditommaso** [comment]
You can control it via the `NXF_PLUGINS_DIR` env variable
